### PR TITLE
asset: move common bootstrap ignition code into a helper type

### PR DESCRIPTION
**Pull request by staebler**
Change the Bootstrap and BootstrapInPlace assets to compose a Common type in order to isolate the in-place specific code.

Add the single-node ignition config target as a subtest of the TestCreatedAssetsAreNotDirty test.

### CODE REVIEW

File: pkg/asset/store/assetcreate_test.go [hunk 1]
```diff
@@ -32,26 +32,57 @@ const userProvidedAssets = `{
...
 		},
 		{
 			name:    "ignition configs",
 			targets: targets.IgnitionConfigs,
+			files:   map[string]string{stateFileName: userProvidedAssets},
+		},
+		{
+			name:    "single node bootstrap-in-place ignition config",
+			targets: targets.SingleNodeIgnitionConfig,
+			files:   map[string]string{"install-config.yaml": singleNodeBootstrapInPlaceInstallConfig},
```
Comment (eranco74) [inline pkg/asset/store/assetcreate_test.go:85]: :+1:

**Review by eranco74 — COMMENTED** [review]
/lgtm

**Comment by staebler** [comment]
@eranco74 The logs from e2e-metal-single-node-live-iso indicate that the must-gather was executed, but the bundle is not included in the job's artifacts. This makes it extremely difficult to diagnose why tests are failing.

**Comment by staebler** [comment]
When testing locally, I do not get any differences in the ignition config generated with or without the changes in this PR.

Let `openshift-install-master` be the installer built from master. Let `openshift-install-PR` be the installer build from this PR.

```
$ mkdir master

$ cat <<EOF > master/install-config.yaml
apiVersion: v1
baseDomain: base.example.com
metadata:
  name: test-cluster
platform:
  none: {}
pullSecret: |
  {
    "auths": {
      "example.com": {
        "auth": "fake-auth"
      }
    }
  }
controlPlane:
  replicas: 1
bootstrapInPlace:
  installationDisk: /dev/sda
EOF

$ openshift-install-master create ignition-configs --dir master
INFO Consuming Install Config from target directory 
INFO Ignition-Configs created in: master and master/auth 

$ cp -r master PR

$ openshift-install-master create single-node-ignition-config --dir master
INFO Consuming Master Ignition Config from target directory 
INFO Consuming Worker Ignition Config from target directory 
INFO Consuming Bootstrap Ignition Config from target directory 
INFO Single-Node-Ignition-Config created in: master and master/auth 

$ openshift-install-PR create signle-node-ignition-config --dir PR
INFO Consuming Master Ignition Config from target directory 
INFO Consuming Worker Ignition Config from target directory 
INFO Consuming Bootstrap Ignition Config from target directory 
INFO Single-Node-Ignition-Config created in: PR and PR/auth 

$ diff master/bootstrap-in-place-for-live-iso.ign PR/bootstrap-in-place-for-live-iso.ign
$ ehco $?
0
```

**Comment by eranco74** [comment]
Looked at the test, it's using a custom image and didn't use the installer built in this PR. 
You can ignore the failure.
I'll look into the must-gather logs
/test e2e-metal-ipi-ovn-ipv6

**Pull request merged**